RoT/Protocol/Challenge_Protocol.md

```diff
@@ -1,5688 +1,2803 @@
 
+# Command Format
 
-    Table 3 Recommended Algorithms for Interoperability
+The following section describes the MCTP message format to support the
+Authentication and Challenge and Attestation protocol.  The Request/Response
+message body describes the Vendor Defined MCTP message encapsulated inside the
+MCTP transport.  This section does not describe the MCTP Transport Header, which
+includes the MCTP Header Version, Destination Endpoint ID and other fields as
+defined by MCTP protocol.  The MCTP message encapsulation is described in
+section 3.2
 
+The MCTP Get Vendor Defined Message Support command enables discovery of what
+Endpoint vendor defined messages are supported.  The discovery identifies the
+vendor organization and defined messages types.  The format of this request is
+described in the MCTP base protocol specification.
 
-<table>
-  <tr>
-   <td><strong>Method</strong>
-   </td>
-   <td><strong>Use</strong>
-   </td>
-  </tr>
-  <tr>
-   <td>X509v3, DER encoding
-   </td>
-   <td>Certificate format
-   </td>
-  </tr>
-  <tr>
-   <td>ECDSA, NIST P256, secp256r1 curve, uncompressed point
-   </td>
-   <td>Digital signing of Certificate
-   </td>
-  </tr>
-  <tr>
-   <td>SHA256
-   </td>
-   <td>Hash algorithm
-   </td>
-  </tr>
-</table>
+For the Cerberus protocol, the following information shall be returned in
+response to the MCTP "Get Vendor Defined Message Support" request:
 
-# Authentication
+*   Vendor ID Format: `0x00`
+*   PCI Vendor ID: `0x1414`
+*   Command Set Version: `0x04`
 
-Authentication is the process used to establish trust in a specific device and prove integrity of the device firmware.  Once a device is authenticated, a secure session can optionally be established to provide confidentiality.  For some devices, a secure session can also be used to enable a cryptographic binding that can be used for additional security or to enable additional functionality.  
 
-A device only needs to support a single session from another endpoint.  In single session devices, the establishment of a new session will supersede and terminate the existing session to permit the new session.  The previous session will be terminated upon receiving an out of session (unencrypted) Get Digests request specifying a requested key exchange.  Get Digests requests received in session (encrypted) or without requesting key exchange will not cause the active session to terminate.  
+## Attestation Protocol Format
 
-## PA-RoT and AC-RoT Authentication
+Messages from a PA-RoT to an AC-RoT will have the following format.
+This is embedded in the structure defined in Section 3.4.
+| Payload  | Description                              |
+|----------|------------------------------------------|
+| 0:0      | Integrity check flag.                    |
+| 7:1      | MCTP message type (always `0x7e`).       |
+| 23:8     | MCTP PCI Vendor ID (always `0x1414`).    |
+| 24:24    | Set if this is a request message.        |
```

Comment:
This conversion lost some detail about this 'rq' bit.  It doesn't really distinguish between request/response messages.

From the original text:
Request Type
   </td>
   <td>This field indicates what type of request is contained in the message.  Messages defined in this specification shall have this bit set to 0.  Setting this bit to 1 provides a mechanism, aside from different vendor IDs, to support a device-specific command set.  Devices that don’t have any additional command support will return an error if this bit is 1.

Reply:
Oops. Shows I've had this wrong the whole time, haha.